system-design/class-diagram/infraestructure/services.puml

```diff
@@ -0,0 +1,30 @@
+@startuml
+package services {
+    package reports {
+        class BorrowStatusReport {
+            - repository: IBorrowRepository
+            + GenerateReport(BorrowStatus borrowStatus, int offset, int limit): Task<string>
+        }
+        BorrowStatusReport *-- ReportsOptions
```

Comment:
I think that the relation is Agregation instead of Composition, im not sure

Reply:
BorrowStatusReport class has been deleted